inference-engine/src/mkldnn_plugin/nodes/mkldnn_non_max_suppression_node.cpp

```diff
@@ -205,12 +195,25 @@ void MKLDNNNonMaxSuppressionNode::execute(mkldnn::stream strm) {
                       });
     }
 
-    const size_t selectedBoxesNum = getChildEdgesAtPort(NMS_SELECTEDINDICES)[0]->getShape().getStaticDims()[0];
+    auto outputMemPtr = getChildEdgesAtPort(NMS_SELECTEDINDICES)[0]->getMemoryPtr();
+    auto& maxOutputDims = outputMemPtr->GetDesc().getShape().getMaxDims();
+    const size_t selectedBoxesNum = maxOutputDims[0];
     const size_t validOutputs = std::min(filtBoxes.size(), selectedBoxesNum);
 
-    int selectedIndicesStride = getChildEdgesAtPort(NMS_SELECTEDINDICES)[0]->getMemory().GetDescWithType<BlockedMemoryDesc>().getStrides()[0];
-    int *selectedIndicesPtr = selected_indices;
-    float *selectedScoresPtr = selected_scores;
+    SizeVector newDims = {validOutputs, maxOutputDims[1]};
+
+    outputMemPtr->redefineDims(newDims);
+
+    if (outputShapes.size() > NMS_SELECTEDSCORES)
```

Comment:
we don't need this check

Reply:
Since we support NMS v5 only, all the extra checks have been removed. Thanks.